internal/pkg/cli/deploy.go

```diff
@@ -206,27 +221,219 @@ func (o *deployOpts) maybeInitWkld(name string) error {
 		return fmt.Errorf("unrecognized workload type %q in manifest for workload %s", workloadType, name)
 	}
 
-	if o.yesInitWkld == nil {
-		confirmInitWkld, err := o.prompt.Confirm(fmt.Sprintf("Found manifest for uninitialized %s %q. Initialize it?", workloadType, name), "This workload will be initialized, then deployed.", prompt.WithFinalMessage(fmt.Sprintf("Initialize %s:", workloadType)))
-		if err != nil {
-			return fmt.Errorf("confirm initialize workload: %w", err)
+	wkldAdder := o.newWorkloadAdder()
+	if err = wkldAdder.AddWorkloadToApp(o.appName, name, workloadType); err != nil {
+		return fmt.Errorf("add workload to app: %w", err)
+	}
+	return nil
+}
+
+type workloadPriority struct {
+	priority  int
+	workloads []string
+}
+type pq []workloadPriority
+
+// These methods satisfy heap.Interface.
+
+// Len returns the length of the data structure.
+func (p pq) Len() int { return len(p) }
+
+// Less returns the lesser of two elements in the array, compared by priority.
+func (p pq) Less(i, j int) bool { return p[i].priority < p[j].priority }
+
+// Swap swaps the positions of two elements in the priority queue.
+func (p pq) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+// Push appends a new element to the back of the underlying array.
+func (p *pq) Push(x any) {
+	*p = append(*p, x.(workloadPriority))
+}
+
+// Pop removes the last element from the array.
+func (p *pq) Pop() any {
```

Comment:
Not a request for this PR, but I feel like ideally we'd have a separate file / package for this and have some unit tests implemented on it!

Reply:
lmao too late I already wrote a generic priority queue